From e43739707804e9800c16e88af8e2087da82f4542 Mon Sep 17 00:00:00 2001
From: Venkateswara Naralasetty <quic_vnaralas@quicinc.com>
Date: Thu, 11 Nov 2021 12:02:27 +0530
Subject: [PATCH] ath11k: use single TCL ring for tx

Using single TCL ring is enough to achieve peak throughput
for QCN6122 and IPQ5018.

Signed-off-by: Venkateswara Naralasetty <quic_vnaralas@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/core.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/net/wireless/ath/ath11k/core.c
+++ b/drivers/net/wireless/ath/ath11k/core.c
@@ -717,7 +717,7 @@ static const struct ath11k_hw_params ath
 		.support_off_channel_tx = false,
 		.is_qdss_support = false,
 		/* In addition to TCL ring use TCL_CMD ring also for tx */
-		.max_tx_ring = DP_TCL_NUM_RING_MAX,
+		.max_tx_ring = 1,
 		.num_vdevs_peers = ath11k_vdevs_peers_ipq5018,
 		.cfr_support = true,
 		.cfr_dma_hdr_size = sizeof(struct ath11k_cfir_dma_hdr),
@@ -799,7 +799,7 @@ static const struct ath11k_hw_params ath
 		.m3_offset = ATH11K_QMI_QCN6122_M3_OFFSET,
 		.qdss_offset = ATH11K_QMI_QCN6122_QDSS_OFFSET,
 		.caldb_offset = ATH11K_QMI_QCN6122_CALDB_OFFSET,
-		.max_tx_ring = DP_TCL_NUM_RING_MAX,
+		.max_tx_ring = 1,
 		.num_vdevs_peers = ath11k_vdevs_peers_ipq5018,
 		.cfr_support = true,
 		.cfr_dma_hdr_size = sizeof(struct ath11k_cfir_enh_dma_hdr),
